File: JavaScript/sort/SelectSort.js
/**
 * 每次循环，选择 [i, A.length - 1] 中最小的值，如果比 A[i] 小，则交换。
 * 
 * @param A: an integer array
 * @return: 
 */
const sortIntegers = function (A) {
  if (A === null || A.length === 0) {
    return;
  }
  
  for (let i = 0; i < A.length - 1; i++) {
    let min = minIndex(A, i + 1);
    if (A[i] > A[min]) {
      let temp = A[i];
      A[i] = A[min];
      A[min] = temp;
    }
  }
}

const minIndex = function(A, start) {
  let min = start;
  for (let i = start; i < A.length - 1; i++) {
    if (A[min] > A[i + 1]) {
      min = i + 1;
    }
  }
  return min;
}

File: Python/sort/SelectSort.py
class Solution:
    """
    @ param A: an integer array
    @ return: nothing
    """
    def sortIntegers(self, A):
        # write your code here        

        for i in range(len(A)-1):
            min_val = A[i]
            temp = i 
            for j in range(i, len(A)-1):
                if min_val > A[j+1]:
                    min_val = A[j+1]
                    temp = j + 1
            A[temp] = A[i]
            A[i] = min_val  
        return A
        